src/remote/action.go

```diff
@@ -52,8 +52,15 @@ func (c *Client) uploadAction(target *core.BuildTarget, isTest, isRun bool) (*pb
 	return command, digest, err
 }
 
-// buildAction creates a build action for a target and returns the command and the action digest digest. No uploading is done.
+// buildAction creates a build action for a target and returns the command and the action digest. No uploading is done.
 func (c *Client) buildAction(target *core.BuildTarget, isTest, stamp bool) (*pb.Command, *pb.Digest, error) {
+	c.buildActionMutex.Lock()
```

Comment:
a little unsure about holding this the whole time since we could end up waiting for non-trivial work (like hashing files or smthn).

Maybe use a sync.Map instead?

Reply:
Yeah will do